NuRadioReco/detector/detector.py

```diff
@@ -1,105 +1,165 @@
-import NuRadioReco.detector.detector_base
-import NuRadioReco.detector.generic_detector
 import json
 import os
+import numpy as np
+import logging
 
-class Detector(object):
-    def __new__(
-            cls,
-            json_filename,
-            source='json',
-            dictionary=None,
-            assume_inf=True,
-            antenna_by_depth=True
-    ):
+from NuRadioReco.detector import detector_base
+from NuRadioReco.detector import generic_detector
+from NuRadioReco.detector.RNO_G import rnog_detector
+
+
+
+def find_path(name):
+    """ Checks for file relative to local folder and detector.py """
+    dir_path = os.path.dirname(os.path.realpath(__file__))  # get the directory of this file
+    filename = os.path.join(dir_path, name)
+    if os.path.exists(filename):
+        return filename
+    else:
+        # try local folder instead
+        if not os.path.exists(name):
+            raise NameError("Can't locate json database file in: {} or {}".format(filename, name))
+        return name
+
+
+def find_reference_entry(station_dict):
+    """
+    Search for the strings "reference_station" or "reference_channel" in the detector description.
+    This is used to determine whether to use the detector_base or generic_detector class.
+    """
+
+    for station in station_dict['stations']:
+        if 'reference_station' in station_dict['stations'][station].keys():
+            return True
+
+    for channel in station_dict['channels']:
+        if 'reference_channel' in station_dict['channels'][channel].keys() or 'reference_station' in \
+                station_dict['channels'][channel].keys():
+            return True
+
+    return False
+
+
+def Detector(*args, **kwargs):
         """
-        Initialize the stations detector properties.
-        This method will check if the JSON file containing the detector description is set up to be used by
-        the DetectorBase or GenericDetector class and cause the correct class to be created.
+        This function returns a detector class object. It chooses the correct class based on the "source" argument.
+        The returned object is of one of these classes:
+
+            - kwargs["source'] == "mongo" -> `NuRadioReco.detector.RNO_G.rnog_detector`
```

Comment:
Can we rename this to "RNOG_mongo". We might have another experiment later that also uses MONGO DB which will make the current naming scheme confusing.